pkg/controllers/state/statenode.go

```diff
@@ -339,11 +338,14 @@ func (in *StateNode) PodLimits() v1.ResourceList {
 }
 
 func (in *StateNode) MarkedForDeletion() bool {
-	// The Node is marked for the Deletion if:
-	//  1. The Node has explicitly MarkedForDeletion
+	// The Node is marked for deletion if:
+	//  1. The Node has the karpenter.sh/disruption:NoSchedule=disrupting taint
 	//  2. The Node has a NodeClaim counterpart and is actively deleting
 	//  3. The Node has no NodeClaim counterpart and is actively deleting
-	return in.markedForDeletion ||
+	_, ok := lo.Find(in.Taints(), func(taint v1.Taint) bool {
+		return v1beta1.DisruptionNoScheduleTaint.MatchTaint(&taint)
```

Comment:
Does match taint match on values and effect? From what I remember, it might just be two of the three?

Reply:
Good catch